client/core/account.go

```diff
@@ -208,24 +209,52 @@ func (c *Core) AccountImport(pw []byte, acct *Account, bonds []*db.Bond) error {
 
 // UpdateCert attempts to connect to a server using a new TLS certificate. If
 // the connection is successful, then the cert in the database is updated.
+// Updating cert for already connected dex will return an error.
 func (c *Core) UpdateCert(host string, cert []byte) error {
-	accountInfo, err := c.db.Account(host)
+	c.connMtx.RLock()
+	dc, found := c.conns[host]
+	c.connMtx.RUnlock()
+	if found && dc.status() == comms.Connected {
+		return errors.New("dex is already connected")
+	}
+
+	acct, err := c.db.Account(host)
 	if err != nil {
 		return err
 	}
 
-	accountInfo.Cert = cert
+	// Stop reconnect retry for previous dex connection first but leave it in
+	// the map so it remains listed incase we need it in the interim.
+	if found {
+		dc.connMaster.Disconnect()
+		dc.acct.lock()
```

Comment:
I see you disconnecting and locking the account, and I see you're leaving the original `dexConnection` in the map, but if we error at `connectDEX` below, this `dexConnection` will be pretty useless.

Reply:
So, we are not supposed to lock the account, correct?

Reply:
Maybe? Right now, it seems like this PR has virtually no net effect. We've moved the disconnect/account lock/book unsub from `connectAccount` to `UpdateCert` and call `connectDEX` instead. But `connectAccount` was just `connectDEX` + disconnect/lock/unsub and adding to `c.conns`. And `connectAccount` is only called one other place, in `(*Core).initialize` where the `dexConnection` will certainly not be in `c.conns`, so no net effect there either. So I guess the only net effect is that the `dexConnection` is added to the map **after** we store the `db.AccountInfo` instead of before.

Reply:
I agree. The new connection is not trade ready(even if we did connect) untill the user re-login. In another fix, we should prepare the new connection to be trade ready, just want to keep the scope of this fix small (i.e prevent disconnecting an already connected DEX).